gpytorch/means/hadamard_multitask_mean.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+
+import torch
+from .multitask_mean import MultitaskMean
+
+
+class HadamardMultitaskMean(MultitaskMean):
+    def forward(self, x, i):
+        """
+        Evaluate the appropriate mean in self.base_means on each input
+        """
+        means = [self.base_means[task](x[i == task]) for task in range(self.num_tasks)]
+
+        res = torch.zeros_like(x)
+        for task in range(self.num_tasks):
+            if len(means[task]) > 0:
+                mean = means[task]
+
+                if res[i == task].dim() == means[0].dim() + 1:
```

Comment:
We shouldn't have to do batch shape checks / unsqueezes here. Every mean should return a `*batch_shape x n_task` tensor, and `res` should be made `*batch_shape x n` (see above comment)